lib/spack/spack/spec.py

```diff
@@ -4772,6 +4773,11 @@ def do_parse(self):
                 pkg = spec.package
                 if hasattr(pkg, 'git'):
                     spec.version.generate_commit_lookup(pkg)
+            # normalize and absolutize 'dev_path'
+            if 'dev_path' in spec.variants:
```

Comment:
```suggestion
            if 'dev_path' in spec.variants and isinstance(spec.variants['dev_path'], vt.AbstractVariant):
```

Maybe create a helper function for it?

Reply:
so you mean a helper function for updating the `dev_path`? so we may also can use this then in https://github.com/spack/spack/blob/fa7189b480d48407bbfafbbf67b1910430a4dc54/lib/spack/spack/concretize.py#L94-L100